### DISCUSSION TITLE
Getting meal plan into home assistant

### DISCUSSION THREAD

**Discussion by CTLS** [mealie-recipes/mealie#454] [Q&A]
It mentions that the information for the days meal plan can be outputted to home assistant as a sensor in the documentation, however there is no explanation how.  Can you provide what you did on the home assistant side to make this work.   I have found a few threads across the internet asking this and I think it would be worth adding to the documentations.

**Comment by hay-kot** [comment]
I'm using node-red to make a get request to the endpoint `/api/meal-plans/today` which will either return the json representation of the recipe scheduled for the day, or nothing. I'm then writing the name attribute to an entity sensor and using that to grab the name in the dashboard. I honestly have no idea how to do this in vanilla home-assistant which is why I haven't included it in the docs. 

If someone wants to do a writeup on how to do this in the vanilla home-assistant config I'd love to include it in the docs.

**Reply by Strider11** [reply]
Would you be willing to share your Node-Red flow for this?

**Reply by hay-kot** [reply]
This receives a web-hook from mealie and writes the name to the created entity. 

```json
[
    {
        "id": "1a6a35de.2ce53a",
        "type": "ha-webhook",
        "z": "12bb1571.511dcb",
        "name": "Mealie Webhook",
        "server": "15e97f32.f16891",
        "version": 1,
        "outputs": 1,
        "webhookId": "mysecretwebhook",
        "outputProperties": [
            {
                "property": "topic",
                "propertyType": "msg",
                "value": "",
                "valueType": "triggerId"
            },
            {
                "property": "payload",
                "propertyType": "msg",
                "value": "",
                "valueType": "data"
            }
        ],
        "x": 760,
        "y": 540,
        "wires": [
            [
                "5f057432.e6398c"
            ]
        ]
    },
    {
        "id": "5f057432.e6398c",
        "type": "json",
        "z": "12bb1571.511dcb",
        "name": "",
        "property": "payload",
        "action": "obj",
        "pretty": false,
        "x": 970,
        "y": 540,
        "wires": [
            [
                "53d1e62e.2938d8"
            ]
        ]
    },
    {
        "id": "53d1e62e.2938d8",
        "type": "ha-entity",
        "z": "12bb1571.511dcb",
        "name": "Meal Today Name",
        "server": "15e97f32.f16891",
        "version": 1,
        "debugenabled": false,
        "outputs": 1,
        "entityType": "sensor",
        "config": [
            {
                "property": "name",
                "value": "mealie_todays_meal"
            },
            {
                "property": "device_class",
                "value": ""
            },
            {
                "property": "icon",
                "value": ""
            },
            {
                "property": "unit_of_measurement",
                "value": ""
            }
        ],
        "state": "payload.name",
        "stateType": "msg",
        "attributes": [],
        "resend": true,
        "outputLocation": "",
        "outputLocationType": "none",
        "inputOverride": "allow",
        "outputOnStateChange": false,
        "outputPayload": "$entity().state ? \"on\": \"off\"",
        "outputPayloadType": "jsonata",
        "x": 1150,
        "y": 540,
        "wires": [
            []
        ]
    },
    {
        "id": "15e97f32.f16891",
        "type": "server",
        "name": "Home Assistant",
        "version": 1,
        "rejectUnauthorizedCerts": true,
        "ha_boolean": "y|yes|true|on|home|open",
        "connectionDelay": true,
        "cacheJson": true
    }
]
```

**Reply by Strider11** [reply]
Thank you!

**Comment by Tommatheussen** [comment]
I just started using Mealie (it's awesome BTW, thanks 10000x @hay-kot 😄), and actually I found a way to get what I want (for now):
Using Home Assistant's REST sensor I am able to get this: 
```
sensor:
  - platform: rest
    resource: http://xxx.xxx.xxx.xxx:xx/api/meal-plans/today
    method: GET
    name: Meal Today
    headers:
      Authorization: Bearer MySuperSecretBearerCode
    value_template: "{{ value_json.name }}"
    json_attributes:
      - slug
````
And in my UI, I created a Picture Entity Card containing the following:
```
type: picture-entity
entity: sensor.meal_today
image: http://xxx.xxx.xxx.xxx:xx/api/meal-plans/today/image
show_name: false
```

So far this works perfectly and I get this result: 
<img width="517" alt="image" src="https://user-images.githubusercontent.com/13683094/122453870-fbc03600-cfaa-11eb-9018-5348959c8088.png">

That being said, I'm actually looking for:
1: A better way to get the image URL inside the REST sensor (so I don't have to duplicate it again)
2: An actual integration, currently nothing from the webhooks is coming in, and having custom messages from Mealie being spoken out from my TTS speakers would just be awesome :D ( I might take a stab at a custom integration myself soonish, if time permits

**Reply by Tommatheussen** [reply]
I created a PR to reflect this in the docs: https://github.com/hay-kot/mealie/pull/549

**Comment by michael-genson** [comment]
This feature is built into Home Assistant now, so you can install the Mealie integration directly through HA
